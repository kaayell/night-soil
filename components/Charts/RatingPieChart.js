import React from 'react'
import {PieChart} from 'react-native-svg-charts'
import {Text, TouchableHighlight, TouchableOpacity, View} from 'react-native'
import style from '../StyleGuide/styles'
import _ from 'lodash'
import {POPPINS} from "../StyleGuide/fonts";

const RATING_COLORS = {
  '1': '#d7ccc8',
  '2': '#a1887f',
  '3': '#8d6e63',
  '4': '#6d4c41',
  '5': '#4e342e'
}

export const RatingPieChart = ({poopData, navigation}) => {
  let grouped = _.groupBy(poopData, 'poopRating')
  let pieData = _.map(grouped, (groupValue, groupKey) => {
    return {
      value: groupValue.length,
      svg: {
        fill: RATING_COLORS[groupKey]
      },
      key: `${Math.random()}`,
    }
  })

  return <TouchableOpacity style={{width: "95%"}} onPress={() => {navigation.navigate('ByRatingDetails', {'poopData': poopData})}}>
    <View style={style.cardStyle}>
      <View style={{height: 40}}>
        <Text style={{fontFamily: POPPINS, color: 'black', textAlign: 'center', paddingTop: 10}}>POOPS BY RATING</Text>
      </View>
      <View style={{flex: 2, flexDirection: 'row', justifyContent: 'space-between'}}>
        <View style={{flex: 2}}>
          <PieChart
            style={{height: 150}}
            data={pieData}
            spacing={0}
            innerRadius={10}
            outerRadius={'95%'}
          />
        </View>
        <View style={{flex: 1, height: "50%"}}>
          <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}><View
            style={{width: 20, height: 10, backgroundColor: RATING_COLORS['1']}}/><Text
            style={{paddingLeft: 10}}>1</Text></View>
          <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}><View
            style={{width: 20, height: 10, backgroundColor: RATING_COLORS['2']}}/><Text
            style={{paddingLeft: 10}}>2</Text></View>
          <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}><View
            style={{width: 20, height: 10, backgroundColor: RATING_COLORS['3']}}/><Text
            style={{paddingLeft: 10}}>3</Text></View>
          <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}><View
            style={{width: 20, height: 10, backgroundColor: RATING_COLORS['4']}}/><Text
            style={{paddingLeft: 10}}>4</Text></View>
          <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}><View
            style={{width: 20, height: 10, backgroundColor: RATING_COLORS['5']}}/><Text
            style={{paddingLeft: 10}}>5</Text></View>
        </View>
      </View>
    </View>
  </TouchableOpacity>
}

export default RatingPieChart