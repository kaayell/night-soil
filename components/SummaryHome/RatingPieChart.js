import React from 'react'
import {PieChart} from 'react-native-svg-charts'
import {Text, TouchableOpacity, View} from 'react-native'
import style from '../StyleGuide/styles'
import {POPPINS} from "../StyleGuide/fonts";
import {pieData, RATING_COLORS} from "../ChartHelpers/pie_chart_helper";

export const RatingPieChart = ({poopData, navigation}) => {
  return (
    <TouchableOpacity style={{width: "95%"}} onPress={() => {
      navigation.navigate('ByRatingDetails', {'poopData': poopData})
    }}>
      <View style={style.cardStyle}>
        <View style={{height: 40}}>
          <Text style={{fontFamily: POPPINS, color: 'black', textAlign: 'left', paddingLeft: 10, paddingTop: 10}}>POOPS BY
            RATING</Text>
        </View>
        <View style={{flex: 2, flexDirection: 'row', justifyContent: 'space-between'}}>
          <View style={{flex: 2}}>
            <PieChart
              style={{height: 150}}
              data={pieData(poopData)}
              spacing={0}
              innerRadius={10}
              outerRadius={'95%'}
            />
          </View>
          <View style={{flex: 1, height: "50%"}}>
            <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}><View
              style={{width: 20, height: 10, backgroundColor: RATING_COLORS['1']}}/><Text
              style={{paddingLeft: 10}}>1</Text></View>
            <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}><View
              style={{width: 20, height: 10, backgroundColor: RATING_COLORS['2']}}/><Text
              style={{paddingLeft: 10}}>2</Text></View>
            <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}><View
              style={{width: 20, height: 10, backgroundColor: RATING_COLORS['3']}}/><Text
              style={{paddingLeft: 10}}>3</Text></View>
            <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}><View
              style={{width: 20, height: 10, backgroundColor: RATING_COLORS['4']}}/><Text
              style={{paddingLeft: 10}}>4</Text></View>
            <View style={{flex: 1, flexDirection: 'row', alignItems: 'center'}}><View
              style={{width: 20, height: 10, backgroundColor: RATING_COLORS['5']}}/><Text
              style={{paddingLeft: 10}}>5</Text></View>
          </View>
        </View>
      </View>
    </TouchableOpacity>)
}

export default RatingPieChart