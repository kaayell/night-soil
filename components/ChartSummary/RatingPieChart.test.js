import React from "react";
import {shallow} from "enzyme";
import RatingPieChart from "./RatingPieChart";

describe('RatingPieChart', () => {
  const poopData = [
    {
      'atWork': true,
      'comments': 'Ieieieie',
      'date': '04-15-2018',
      'durationMinutes': '3',
      'poopRating': 3,
      'salary': '23',
    },
    {
      'atWork': true,
      'date': '04-14-2018',
      'durationMinutes': '5',
      'poopRating': 5,
      'salary': '30',
    },
    {
      'atWork': true,
      'date': '04-14-2018',
      'durationMinutes': '3',
      'poopRating': 5,
      'salary': '30',
    },
    {
      'atWork': true,
      'date': '04-15-2018',
      'durationMinutes': '1',
      'poopRating': 2,
      'salary': '30',
    },
    {
      'atWork': false,
      'date': '04-15-2018',
      'durationMinutes': '1',
      'poopRating': 3,
      'salary': '30',
    },
  ]

  it('should render', () => {
    const wrapper = shallow(<RatingPieChart poopData={poopData}/>)
    expect(wrapper).toMatchSnapshot()
  });

});