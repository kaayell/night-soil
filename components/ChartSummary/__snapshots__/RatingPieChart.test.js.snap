// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingPieChart should render 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "width": "95%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 200,
        "justifyContent": "center",
        "marginTop": 20,
        "shadowColor": "#9AC0CD",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 5,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "height": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontFamily": "poppins-regular",
            "paddingLeft": 10,
            "paddingTop": 10,
            "textAlign": "left",
          }
        }
      >
        POOPS BY RATING
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 2,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <PieChart
          data={
            Array [
              Object {
                "key": "#a1887f",
                "svg": Object {
                  "fill": "#a1887f",
                },
                "value": 1,
              },
              Object {
                "key": "#8d6e63",
                "svg": Object {
                  "fill": "#8d6e63",
                },
                "value": 2,
              },
              Object {
                "key": "#4e342e",
                "svg": Object {
                  "fill": "#4e342e",
                },
                "value": 2,
              },
            ]
          }
          height={100}
          innerRadius={10}
          outerRadius="95%"
          padAngle={0.05}
          renderDecorator={[Function]}
          sort={[Function]}
          spacing={0}
          style={
            Object {
              "height": 150,
            }
          }
          valueAccessor={[Function]}
          width={100}
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": "50%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#d7ccc8",
                "height": 10,
                "width": 20,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "paddingLeft": 10,
              }
            }
          >
            1
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#a1887f",
                "height": 10,
                "width": 20,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "paddingLeft": 10,
              }
            }
          >
            2
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#8d6e63",
                "height": 10,
                "width": 20,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "paddingLeft": 10,
              }
            }
          >
            3
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#6d4c41",
                "height": 10,
                "width": 20,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "paddingLeft": 10,
              }
            }
          >
            4
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#4e342e",
                "height": 10,
                "width": 20,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "paddingLeft": 10,
              }
            }
          >
            5
          </Text>
        </View>
      </View>
    </View>
  </View>
</TouchableOpacity>
`;
